### PR TITLE
Use pure `pyproject.toml` setup for packaging

### PR DESCRIPTION
**Pull request by bittner**
Phases out `setup.py` and `requirements.txt` in a quest to modernize packaging.

This PR also drops support for Python 3.6, which is not supported by newer versions of setuptools required to read all values out of `pyproject.toml` (e.g. name and authors).

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,12 +33,9 @@ jobs:
         exclude:
         - { django-version: '3.2', python-version: '3.11' }
         - { django-version: '3.2', python-version: '3.12' }
-        - { django-version: '4.1', python-version: '3.6' }
         - { django-version: '4.1', python-version: '3.7' }
         - { django-version: '4.1', python-version: '3.12' }
-        - { django-version: '4.2', python-version: '3.6' }
         - { django-version: '4.2', python-version: '3.7' }
```
Comment (kingbuzzman) [inline .github/workflows/test.yml:38]: Might as well get rid of this too: https://endoflife.date/python as of june of last year.

File: README.rst [hunk 2]
```diff
@@ -25,7 +25,7 @@ Version Support
 ---------------
 
 *behave-django* is `tested against`_ the officially supported combinations of
-Python and Django (Django 3.2, 4.1, 4.2, 5.0 on Python 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12).
+Python and Django (Django 3.2, 4.1, 4.2, 5.0 on Python 3.7, 3.8, 3.9, 3.10, 3.11, 3.12).
```
Comment (kingbuzzman) [inline README.rst:28]: heh. django 3.2 has a month and a half left of life https://endoflife.date/django

File: requirements.txt
Comment (kingbuzzman) [inline requirements.txt:1]: YES!

File: tox.ini [hunk 3]
```diff
@@ -8,7 +8,7 @@ envlist =
     isort
     pylint
     # Python/Django combinations that are officially supported
-    py3{6,7,8,9,10}-django32
+    py3{7,8,9,10}-django32
```
Comment (kingbuzzman) [inline tox.ini:11]: Any particular reason for tox now? Figured with github actions, this was moot.

File: .github/workflows/test.yml (hunk 1 above)
Comment (bittner) [reply .github/workflows/test.yml:38]: [Django 3.2 LTS](https://pypi.org/project/Django/3.2.24/) supports Python 3.6 through 3.10 according to the metadata on PyPI. It's more for that that it makes sense to keep Python 3.7 around.

But we can certainly drop the obsolete Django 4.1. See #157

File: README.rst (hunk 2 above)
Comment (bittner) [reply README.rst:28]: I'd love to push a release to PyPI before Django 3.2 is obsoleted. There's still time to drop it afterwards.

The real problem is the dependency of behave-django [on a development release](https://github.com/behave/behave-django/blob/main/pyproject.toml#L52) of Behave, though. It makes use in air-gapped environments very difficult if not entirely impossible.

File: tox.ini (hunk 3 above)
Comment (bittner) [reply tox.ini:11]: Tox is a tool that [belongs to the development layer](https://pythonbiellagroup.gitlab.io/testinginpython/en/tutorial/automazione.html#application-development-and-deployment-automation). It abstracts the details of running tests and linting away. Last but not least, it allows to run all the things locally, without the automation component.

I guess it's here to stay. Is there any good reason to get rid of it?

**Pull request merged**